src/sidebar/components/NotebookView.js

```diff
@@ -37,11 +38,20 @@ function NotebookView({ loadAnnotationsService }) {
   const lastPaginationPage = useRef(1);
   const [paginationPage, setPaginationPage] = useState(1);
 
+  const [hasLoadError, setHasLoadError] = useState(false);
+
+  // Load all annotations in the group, unless there are more than 5000
+  // of them: this is a performance safety valve.
+  const maxResults = 5000;
+
+  const onLoadError = error => {
+    if (error.message?.includes('Results size exceeds')) {
```

Comment:
I would suggest defining a specific `Error` subclass for this error. The risk with testing the `message` string is that someone changes it (as `message` is not in general intended to be machine-readable) and doesn't realize that it is important.

For example:

```js
class ResultSizeError extends Error {
  constructor(limit) {
    super(`Results size exceeds ${limit}`);
  }
}
```

Then:

```js
if (error instanceof ResultSizeError) {
  ...
}
```

A variation on the above is to define a more general error subclass for use in some context and then add a property (eg. `type`) with a value that is specifically intended to be machine-readable.

As a side benefit, this provides somewhere to hang additional error properties if you need them.